runtime/Go/antlr/dfa.go

```diff
@@ -5,7 +5,7 @@ import "sort"
 type DFA struct {
 	atnStartState DecisionState
 	decision      int
-	_states       map[string]*DFAState
+	states        map[string]*DFAState
```

Comment:
At this point I would like to ask a general question on how to go forward in the long term. In my opinion we could potentially agree on som goals:

1. turn all tests green and implement some "ANTLR contract"
2. make go code idiomatic
3. make it performant
4. keep track with further innovations of ANTLR

IMHO point 2 is important for users who import the runtime API and maybe lookup code in order to quickly understand behaviour (the last one is one of Go's strong suits). Also I think 2 is important for 3, although 3 should not have any importance right now. 

The question I'd like to discuss is the relation of 2 and 4. How far can we and should we deviate from the code base? I don't have a complete overview of all the things yet, but if I understand Peter correctly, a lot of the code comes from some (semi-)automatic transformation from Java+JavaScript to Go. Some structure and naming convention that would conflict goal 2 may be kept in order to make tracking upstream changes (in the Java target) easier. 

I am asking this general question at this line of code, because this could be a simple example where we may (as I said, I am lacking sufficient overview as of now) loose some bit of information that might bite us in the mid/long term. 

I don't have an opinion on this big picture question, yet. I'd be happy as well when we just keep the exported signatures as they are (and go-ify them), maybe unexport some that needn't be exported and reduce the private code to the essence of what idiomatic, highly optimized go is. But I'd like to have an educated understanding of any middleground we may want to keep, before the ANTLR Go target goes into maintenance mode in the long term.